### PR TITLE
feat: Make private URLs easier to copy-paste from web app

### LINKED ISSUE #12996: Make private URLs easier to copy-paste from web app
### Is your proposal related to a problem?

Frustrating to have to click into a few things to get to private URL for any given event for scheduling.

### Describe the solution you'd like

When on https://app.cal.com/event-types ideally there is another copy button next to the normal one for copying and using "private URLs."

image

### Describe alternatives you've considered

n/a
### Additional context

n/a

### Requirement/Document

n/a

---
##### House rules
- If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  - For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  - Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.

### PR DESCRIPTION
**Pull request by riddhesh-mahajan**
## What does this PR do?

Adds private link copy icon on /event-types page. Icon will be visible if private url is enabled for a specific event.

Fixes #12996 

Private link copy

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How should this be tested?

- Edit event
- Go to Advanced menu
- Enable private url
- Go back to /event-types page

### CODE REVIEW

**Review by PeerRich — DISMISSED** [review]
I wonder if an incognito icon would make sense here:

CleanShot 2024-01-03 at 17 08 58@2x

svg: https://github.com/calcom/cal.com/assets/8019099/36237e55-40ab-46b3-9fd3-5d8f517a588a
incognito

**Comment by PeerRich** [comment]
> I wonder if an incognito icon would make sense here:
> 
> CleanShot 2024-01-03 at 17 08 58@2x
> 
> svg: [calcom/cal.com/assets/8019099/36237e55-40ab-46b3-9fd3-5d8f517a588a](https://github.com/calcom/cal.com/assets/8019099/36237e55-40ab-46b3-9fd3-5d8f517a588a) incognito

or the venecian mask from lucide icons:
https://lucide.dev/icons/venetian-mask
venetian-mask

File: apps/web/pages/event-types/index.tsx [hunk 1]
```diff
@@ -465,6 +469,28 @@ export const EventTypeList = ({
                                   }}
                                 />
                               </Tooltip>
+
+                              {isPrivateURLEnabled && (
+                                <Tooltip content={t("copy_link")}>
+                                  <Button
+                                    color="secondary"
+                                    variant="icon"
+                                    StartIcon={LinkIcon}
```
Comment (PeerRich) [inline apps/web/pages/event-types/index.tsx:478]:
```suggestion
                                    StartIcon={VenetianMask}
```

File: apps/web/pages/event-types/index.tsx [hunk 2]
```diff
@@ -465,6 +469,28 @@ export const EventTypeList = ({
...
+                                    }}>
+                                    <Lock
+                                      style={{
+                                        maxHeight: "0.6rem",
+                                        position: "absolute",
+                                        left: "14px",
+                                        top: "20px",
+                                      }}
+                                    />
+                                  </Button>
```
Comment (PeerRich) [inline apps/web/pages/event-types/index.tsx:491]:
```suggestion
                                    }}/>
```

File: apps/web/pages/event-types/index.tsx [hunk 3]
```diff
@@ -66,6 +67,7 @@ import {
   Trash,
   Upload,
   Users,
+  Lock,
```
Comment (PeerRich) [inline apps/web/pages/event-types/index.tsx:70]:
```suggestion
  VenetianMask,
```

File: apps/web/pages/event-types/index.tsx [hunk 4]
```diff
@@ -465,6 +469,19 @@ export const EventTypeList = ({
...
+                                  <Button
+                                    color="secondary"
+                                    variant="icon"
+                                    StartIcon={VenetianMask}
+                                    onClick={() => {
+                                      showToast(t("private_link_copied"), "success");
+                                      navigator.clipboard.writeText(placeholderHashedLink);
+                                    }}/>
+                                </Tooltip>
+                              )}
```
Comment (PeerRich) [inline apps/web/pages/event-types/index.tsx:484]:
```suggestion
                              {isPrivateURLEnabled && (
                                      <Tooltip content={t("copy_link")}>
                                        <Button
                                          color="secondary"
                                          variant="icon"
                                          StartIcon={VenetianMask}
                                          onClick={() => {
                                            showToast(t("private_link_copied"), "success");
                                            navigator.clipboard.writeText(placeholderHashedLink);
                                          }}
                                        />
                                      </Tooltip>
                                    )}
```

**Comment by riddhesh-mahajan** [comment]
> I wonder if an incognito icon would make sense here:

venecian mask icon would be better

**Review by PeerRich — APPROVED** [review]

**Pull request merged**